basic_module/distribution/src/impls.rs

```diff
@@ -13,3 +13,54 @@
 //
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::core::OpenBlockHandler;
+use crate::error::Error;
+use crate::{
+    calculate_pending_rewards_of_the_term, current_term_id, get_banned_validators,
+    get_block_header, get_intermediate_rewards, get_key,
+    last_term_finished_block_num,
+};
+use ckey::Ed25519Public as Public;
+use ctypes::BlockHash;
+use std::collections::BTreeMap;
+
+#[allow(dead_code)]
+struct Distribution {
+    block_number: u64,
+    block_author: Public,
+}
+
+impl OpenBlockHandler for Distribution {
+    fn distribute_when_open_block(
```

Comment:
I think what you should do in `open_block` is recording some necessary information. You should execute distribution when executing the corresponding transaction.

Reply:
ditto in `close_block`